### PR TITLE
[Closes #117] Replaces `kerla_utils::bitmap` with `bitvec::BitArray`

### LINKED ISSUE #117: Replace `kerla_utils::bitmap` with `bitvec::array::BitArray`
I noticed that [bitvec](https://docs.rs/bitvec/latest/bitvec/array/struct.BitArray.html) provides sufficient features for us as a bitmap container. This is a refactoring issue to replace all bitmaps in Kerla with bitvec.

### PR DESCRIPTION
**Pull request by michalfita**
## Description
This is actually pretty straightforward change and in result the burden of in-house maintenance is reduced.
It closes #117.

## Pre-Submission Checklist

When you submit a PR, please make sure your PR satisfies the following checklist:

- [x] I assert this contribution was authored by me.
- [x] I license this contribution under the license of this project.
- [x] The PR title describes your changes briefly and uses the present tense, without a trailing full stop.
- [x] The PR description describes the reason why we need the change.
- [x] This is an isolated change. No multiple changes and no unrelated changes are included.
- [ ] Fixed all `cargo clippy` warnings.
- [ ] Applied `rustfmt`.

### CODE REVIEW

File: libs/kerla_utils/id_table.rs [hunk 1]
```diff
@@ -1,21 +1,22 @@
-use crate::bitmap::BitMap;
+use core::mem::size_of;
+use bitvec::prelude::*;
 
-pub struct IdTable<const SZ: usize>(BitMap<SZ>);
+pub struct IdTable<const BIT_LENGTH: usize>(BitArray<[usize; BIT_LENGTH], LocalBits>); // const expr arithmetic unstable
```
Comment (nuta) [inline libs/kerla_utils/id_table.rs:4]:
```suggestion
pub struct IdTable<const BIT_LENGTH: usize>(BitArray<[usize; BIT_LENGTH], LocalBits>);
```

I'm not sure what you want to point out here. Do you mean we need a feature gate to use this expression?

Comment (michalfita) [reply libs/kerla_utils/id_table.rs:4]: @nuta I wanted to calculate compile time the size of the BitArray from the number of required bits, but the feature isn't good enough to use even with the nightly at the moment. I can remove that comment...

Comment (nuta) [reply libs/kerla_utils/id_table.rs:4]: Makes sense. Rust's const fn is still developing so let's leave this comment. Thanks!

**Review by nuta — APPROVED** [review]

**Comment by nuta** [comment]
Thanks again @michalfita!

**Pull request merged**